### DISCUSSION TITLE
Windows Portable version

### DISCUSSION THREAD

**Discussion by Korb** [edrlab/thorium-reader#2740] [Ideas]
I propose to consider the possibility of publishing in Releases a **portable** version of the Thorium Reader for Windows, in addition to Thorium.Setup.x.x.x.exe.

**Comment by danielweck** [comment]
hello, what would be the advantages for the end user? are there problems with the existing installer that this would solve? (currently, the installer offers per-user and global setup options)

**Comment by Korb** [comment]
The advantages are the same as those of a portable application over an installed one. When an application does not need to install drivers, services, change the Windows File Manager context menu, add its modules to other applications, I prefer to use portable versions of applications. This reduces the number of changes in the registry, leaves less garbage in system folders after deleting the application, and speeds up the process of reinstalling the OS _(installation and configuration of applications is reduced to specifying the old folder for storing start menu shortcuts and writing mappings of applications to file extensions)_.

**Comment by panaC** [comment]
https://www.electron.build/nsis#portable

**Comment by Svetloslav** [comment]
Not to mention, that true portability, as in PortableApps project, makes it easy to port huge book libraries, together with the reader (Thorium) between many workstations (windows PCs). Also program updates, reading progress and taking of notes, etc., those can be all centralized by residing in the same directory on  an external medium (USB flash drive/disk). This can be the only place one should focus efforts and than it all will be available on any terminal at hand. Said benefits are impossible to ignore, for the installation of software, other than the hardware drivers, is nothing but a barbarian doctrine.
